### PR TITLE
Df ist output

### PR DESCRIPTION
**Pull request by dflanderka**

### CODE REVIEW

File: src/input/type_output.cc [hunk 1]
```diff
@@ -3,6 +3,8 @@
  */
 
 #include "input/type_output.hh"
+#include "input/type_repository.hh"
+#include "rev_num.h"
```
Comment (jbrezmorf) [inline src/input/type_output.cc]: This include is forbidden for every  source else then main.cc. 
All information have to be passed through appropriate method of OutputJSON.

File: src/input/type_output.hh [hunk 2]
```diff
@@ -11,6 +11,7 @@
 
 #include "input/type_base.hh"
 #include "input/type_record.hh"
+#include "input/type_generic.hh"
 #include <boost/regex.hpp>
```
Comment (jbrezmorf) [inline src/input/type_output.hh]: Can we remove this include?

File: unit_tests/input/type_output_test.cpp [hunk 3]
```diff
@@ -92,24 +92,10 @@ TEST(OutputTypeTypeBase, record_output_test) {
...
-    cout << OutputLatex(&main) << endl;
+    cout << OutputText(&main) << endl << endl;
 
     cout << "## " << "OutputJSONMachine printout" << endl;
-
-    cout << OutputJSONMachine(&main) << endl;
+    cout << OutputJSONMachine() << endl;
 }
 
 /*TEST(OutputTypeAbstractRecord, abstract_record_test) {
```
Comment (jbrezmorf) [inline unit_tests/input/type_output_test.cpp]: Why these tests are commented. Try to uncomment and possibly fix them.

File: src/input/type_output.hh (hunk 2 above)
Comment (dflanderka) [inline src/input/type_output.hh]: No, we use boost::regex in inner class OutputJSONMachine::RewriteRule

Comment (jbrezmorf) [inline src/input/type_output.hh]: RewriteRule is used only in escape_description and is mere pair. So, we can remove class RewriteRule and use 
static vector of pairs in the method OutputJSONMachine::escape_description.

File: src/main.cc [hunk 4]
```diff
@@ -112,23 +112,32 @@ void Application::split_path(const string& path, string& directory, string& file
     }
 }
 
+
+std::map<string, string> Application::get_rev_num_data() {
```
Comment (jbrezmorf) [inline src/main.cc]: Why using map where we can use a struct?
Declare appropriate struct in type_output.hh.

**Pull request merged**